internal/dbmodel/sql/postgres/020_add_ssh_keys_table.up.sql

```diff
@@ -0,0 +1,14 @@
+-- Add the ability to store users' SSH public keys
+
+CREATE TABLE ssh_keys (
+    id SERIAL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
+    updated_at TIMESTAMP WITH TIME ZONE,
+    public_key BYTEA NOT NULL,
```

Comment:
i think juju displays hexadecimal md5 key signature when you run `juju ssh-keys` so i'm wondering if it would be worth storing that as well so that we can do the lookup

Reply:
but if it would enable us to do the lookup based on key fingerprint without requiring the user to specify the username explicitly..

Reply:
But I believe it would be incorrect to query the keys without the user specifying a username. It's a bit contrived but imagine I gave you my SSH key pair (or in the incredibly slim chance you happen to generate the same key). You could also upload the public key to the controller and then on login there are 2 keys that match.

It's the same principle as basic auth but without requiring a username. I don't know what the guarantees around uniqueness of SSH keys are (how long they are, their entropy, etc) but I'm fairly confident it would be wrong to query for keys without the "username" of the user attempting to login.